index.js

```diff
@@ -62,9 +62,9 @@ module.exports.lintText = (str, opts) => {
 	if (opts.filename) {
 		const filename = path.relative(opts.cwd, opts.filename);
 		const isIgnored = multimatch(filename, opts.ignores).length > 0;
-		const isGitIgnored = !optionsManager.getGitIgnoreFilter(opts)(opts.filename);
+		const isGitIgnored = globby.gitignore.sync({cwd: opts.cwd, ignore: opts.ignores});
 
-		if (isIgnored || isGitIgnored) {
+		if (isIgnored || isGitIgnored(opts.filename)) {
```

Comment:
I prefer conditions like this one to be symmetric, so deciding on either

```js
if (fn1(filename) || fn(filename)) {

}
```

**or**

```js
var one = fn1(filename);
var two = fn2(filename);

if (one || two) {

}
```

would help readability here. Doing the execution late could help save some io in some cases but without a proof this has a meaningful impact on performance I'd opt for higher readability (additional binding to intermediary variables).

Reply:
Actually I made this change from your code because it was really confusing to me why `multimatch(filename, opts.ignores).length > 0` was wrap into a function. It took me several minutes to figurate out the function wrapper was useless.

It might looks more elegant or symmetrical to have `if (isIgnored(opts.filename) || isGitIgnored(opts.filename))` instead of `if (isIgnored || isGitIgnored(opts.filename))` but in terms of cognitive load it's really confusing to wrap something in a function to call it on the next line.

It's always a good idea to improve the formatting or splitting the code for readability, but I'm not convinced that changing the behavior of the code (like wrapping in a function) for readability purpose is so good.

Reply:
We are on the same page then with 

```js
const isIgnored = match(...args1);
const isGitIgnored = match(...args2)(...args3);

if (isIgnored || isGitIgnored) {

}
```

being the best option.

Reply:
In that situation you would systematically call `globby.gitignore.sync(...args2)(...args3)`. While with the current solution it's called only if `isIgnored` is falsy.

Calling it has a performance cost as it requires to read the `.gitignore` file.
So there is no need to do that if the file is ignored anyway and `isIgnored` is falsy.

I don't think it's a good idea to make the code behave differently and read a file on the disk, in order to make an `if` look symmetrical.

Reply:
Sure, that would be true if for`globby.gitignore.sync(...args2)(...args3)` the second invocation `fn(...args3)` triggered the IO/side effects. 

In the current implemenation over at `globby` [this work](https://github.com/sindresorhus/globby/blob/master/gitignore.js#L84) already is done during the first invocation `fn(...args2)`. 

A performance optimization with impact would be:

```js
if (match(...args) || globby.gitignore.sync(...args2)(...args3)) {

}
```

A bit less terse / maybe more readable

```js
if (match(...args)) {
  return {...};
}

const isGitIgnored = globby.sync(...args2);

if (isGitIgnored(...args3)) {
  return {...};
}
```

Reply:
I change it for the following:
```js
if (multimatch(filename, opts.ignores).length > 0 ||
  globby.gitignore.sync({cwd: opts.cwd, ignore: opts.ignores})(opts.filename)) {

}
```

But really, code is more readable and easy to understand if it simply does what its meant for. Changing it in a convoluted way for symmetrical reasons just add to the cognitive load of understanding it.

Reply:
We'll have to disagree on this and recognize cognitive load might be caused by different things for each of us. 

One additional thought: an early return might express your **intent** to save work if possible a bit better.

I am fine with the intended change regardless.